AstarteDeviceSDKCSharp/Device/AstarteDevice.cs

```diff
@@ -221,10 +221,16 @@ public void SetAlwaysReconnect(bool alwaysReconnect)
         public async Task Connect()
         {
 
+            if (!_pairingHandler.IsCertificateAvailable())
+            {
+                await _pairingHandler.RequestNewCertificate();
+                _initialized = false;
+            }
+
```

Comment:
Perhaps this piece of code, should be moved after the init step, because if the transport is not yet initialized, it cannot request a new certificate.

``` C#
 if (!_initialized)
{
   await Init();
   _initialized = true;
}

 if (!_pairingHandler.IsCertificateAvailable())
{
    await _pairingHandler.RequestNewCertificate();
    _initialized = false;
}
```

Reply:
I put this line before init because I need to check if device.key is expired first. Init function read device.key from file if device.key PublishAsync method throw exception and agine call Connect method. Based on my test, this is a better way to regenerate a device.key.